### PR TITLE
Support execution of programs for JVM compilers

### PR DESCRIPTION
**Pull request by supergrecko**
This patch contains the bare minimum to get Kotlin and Java to execute in a sandboxed environment using the infrastructured added in 68a687decb51142c75ee99a06374702487c0e3b3

Due to JVM internals (it spawning about 13 threads + main), the current pid limit does not suffice. 14 pids is the minimum I've been able to make it work with, but I hope to reduce this number.

### CODE REVIEW

File: etc/nsjail/sandbox.cfg [hunk 1]
```diff
@@ -28,7 +28,7 @@ cgroup_net_cls_parent: "ce-sandbox"
 cgroup_cpu_parent: "ce-sandbox"
 
 cgroup_mem_max: 209715200 # 200 MiB
-cgroup_pids_max: 4
+cgroup_pids_max: 14
```
Comment (supergrecko) [inline etc/nsjail/sandbox.cfg:31]: This is the minimum amount of pids I was able to make the executors run. Started discussion on CE's Discord regarding this.

File: lib/compilers/java.js [hunk 2]
```diff
@@ -103,6 +103,29 @@ export class JavaCompiler extends BaseCompiler {
         ];
     }
 
+    async handleInterpreting(key, executeParameters) {
+        executeParameters.args = [
+            '-Xss136K', // Reduce thread stack size
+            '-XX:CICompilerCount=2', // Reduce JIT compilation threads. 2 is minimum
+            '-XX:-UseDynamicNumberOfCompilerThreads',
+            '-XX:-UseDynamicNumberOfGCThreads',
+            '-XX:+UseSerialGC', // Disable parallell/concurrent garbage collector
```
Comment (supergrecko) [inline lib/compilers/java.js:112]: These are all options used to reduce the memory and processes/threads used by the JVM.

File: etc/nsjail/sandbox.cfg (hunk 1 above)
Comment (mattgodbolt) [reply etc/nsjail/sandbox.cfg:31]: Thanks for commenting this. It hurts me this isn't a power of two ;) We need to work out on our side if having multiple executions, each having 14 (or 16 😁 ) threads is a DoS potential. I think _not_ as the `cgroup_cpu_ms_per_sec` should limit complete CPU takeover.

File: etc/config/kotlin.amazon.properties [hunk 3]
```diff
@@ -9,7 +9,8 @@ postProcess=
 options=
 supportsBinary=false
 needsMulti=false
-supportsExecute=false
+supportsExecute=true
+interpreted=true
```
Comment (mattgodbolt) [inline etc/config/kotlin.amazon.properties:13]: Neat; I'm glad the new feature was somewhat useful...

**Review by mattgodbolt — APPROVED** [review]
On balance I think this is going to be OK. Thanks so much for this!

**Pull request merged**